File: archive/p/python/capitalize.py
import sys


def capitalize(input):
    if len(input) > 0:
        print(input[0].capitalize() + input[1:])


if __name__ == '__main__':
    if(len(sys.argv) == 1 or len(sys.argv[1]) == 0):
        print('Usage: please provide a string')
    else:
        capitalize(sys.argv[1])

File: archive/p/pascal/capitalize.pas
program Capitalise(input, output, stdErr);
(* Accept a String from commandline, convert first character to upper case
   Pascal Strings begins with 1, not 0
*)
var
buf: string;
begin
  buf := paramStr(1);
  if buf = ''  then
    writeln('Usage: please provide a string')
  else
    buf[1] := UpCase(buf[1]);
    writeln(buf);
end.
